R/subset.R

```diff
@@ -58,14 +61,36 @@ subset.ddpcr_plate <- function(x, wells, samples, ...) {
       plate_meta(x) %>%
       dplyr::filter_(~ sample %in% samples) %>%
       .[['well']]
+  } else if (!missing(targets_ch1) | !missing(targets_ch2)) {
+    wells <-
+      plate_meta(x) %>%
+      .[['well']]
   } else {
     return(x)  # if no arguments, just return the same plate
   }
   
+  if (!missing(targets_ch1)) {
+    wells_from_targets <-
+      plate_meta(x) %>%
+      dplyr::filter_(~ target_ch1 %in% targets_ch1) %>%
```

Comment:
If invalid targets are given, the plate is returned unchanged. I did this to stay consistent with previous behavior. subset(plate, samples = 'noSample') returns plate as well.  However, I'm not sure, if this is the best way. Sample and targets are usually some abbreviations, that are prone to typing errors.  At warning might make analysis safer, e.g.:

    # if no valid wells were given, don't do anything
    if (sum(wells %in% (x %>% wells_used)) == 0) {
       warn_msg("No valid wells selected. Returning plate unchanged.")
       return(x)
    }